openff/interchange/components/_packmol.py

```diff
@@ -706,15 +706,25 @@ def pack_box(
                     packmol_path,
                     stdin=file_handle,
                     stderr=subprocess.STDOUT,
-                ).decode("utf-8")
+                )
             except subprocess.CalledProcessError as error:
-                if "173" in str(error):
-                    raise PACKMOLRuntimeError from error
+                # Custom error codes seem to start at 170
+                # https://github.com/m3g/packmol/blob/v20.15.1/src/exit_codes.f90#L13-L16
+                open("packmol_error.log", "w").write(error.stdout.decode("utf-8"))
```

Comment:
Is there a reason to write this to a file instead of just dumping to stderr? `print(error.stdout.decode("utf-8"), file=sys.stderr)` would be about as terse and avoid having to warn about a temporary directory down below. Or you could write it into a `StringIO` and include it in the exception itself.

Reply:
Mostly that this is the easiest way I could think of dumping the error log in a way that the user could inspect later (but does require opting in to `working_directory='.', retain_working_files=False`). Also, it can be fairly long, like a few thousand lines for realistic systems. IIUC dumping to STDERR would display the same information to the user but potentially get stuff merged in with other errors (and maybe warnings?) which could be noisier

Reply:
Sounds good. As a user I think I'd slightly prefer to have it all in one place, but I definitely see the tradeoff and think this is a reasonable choice too!